source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+								{kind: 'moon.IntegerPicker', name: 'minute', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.minute, disabled: this.countdown},
+								{name: 'minuteLabel', content: this.minuteText, classes: 'moon-date-picker-label moon-divider-text'}
+							]}
+						);
+						break;
+					case 's':
+						this.createComponent(
+							{name: 'secondPicker', classes: 'moon-date-picker-wrap', components:[
+								{kind: 'moon.IntegerPicker', name: 'second', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.second, disabled: this.countdown},
+								{name: 'secondLabel', content: this.secondText, classes: 'moon-date-picker-label moon-divider-text'}
+							]}
+						);
+						break;
+					}
+				}
+			} else {
+				this.createComponent(
+					{name: 'hourPicker', classes: 'moon-date-picker-wrap', components:[
+						{kind: 'moon.IntegerPicker', name: 'hour', min: 0, max: 23, wrap: true, value: this.valueArray.hour, disabled: this.countdown},
+						{name: 'hourLabel', content: this.hourText, classes: 'moon-date-picker-label moon-divider-text'}
+					]}
+				);
+				this.createComponent(
+					{name: 'minutePicker', classes: 'moon-date-picker-wrap', components:[
+						{kind: 'moon.IntegerPicker', name: 'minute', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.minute, disabled: this.countdown},
+						{name: 'minuteLabel', content: this.minuteText, classes: 'moon-date-picker-label moon-divider-text'}
+					]}
+				);
+				this.createComponent(
+					{name: 'secondPicker', classes: 'moon-date-picker-wrap', components:[
+						{kind: 'moon.IntegerPicker', name: 'second', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.second, disabled: this.countdownSS},
+						{name: 'secondLabel', content: this.secondText, classes: 'moon-date-picker-label moon-divider-text'}
+					]}
+				);
+			}
+			this.render();
+		},
+
+		/**
+		* @private
+		*/
+		currentTemplate: function () {
+			if (this.template) {
+				var orderingArr = this.template.toLowerCase().split('');
+				var o,f,l;
+				for (f = 0, l = orderingArr.length; f < l; f++) {
+					o = orderingArr[f].toLowerCase();
+					//only accepts the hour, minute, second template values 'h', 'm' and 's'
+					if (this.templateArray.indexOf(o) < 0 && (o == 'h' || o == 'm' || o == 's' )) {
+						this.templateArray.push(o);
+					}
+				}
+				if (!this.templateArray.length) {
+					this.templateArray.push('h','m','s');
+				}
+				return;
+			}
+			this.templateArray.push('h','m','s');
+		},
+
+		/**
+		* @private
+		*/
+		templateChanged: function () {
+			this.templateArray = [];
+			this.deletePickers();
+			this.currentTemplate();
+			this.setupPickers();
+			this.setupArray();
+			this.setValue(this.formatValue());
+		},
+
+		/**
+		* @private
+		*/
+		setupArray: function () {
+			if (this.template.search('h') < 0)
```

Comment:
I think using `indexOf` would be more performant here: http://jsperf.com/exec-vs-match-vs-test-vs-search/11.